### ISSUE TITLE
`repr()` is misbehaving

### ISSUE THREAD

**Issue by yunruse** [yunruse/Noether#5]
I think I've leant on `rich` too much, because without it, the following is a bit borked:

```
>>> from noether import meter
>>> meter
Unit(value=1, stddev=None, dim=length  # length, length, distance, height, width, breadth, depth)
>>> conf.display_repr_code
False
```

This should only happen when `display_repr_code` is True.
Additionally, the comment should _not_ display when it is True!

This should be enforced by some unit testing, I think.

**Comment by yunruse** [comment]
`repr(Dimension)` and `str(Dimension)` seem swapped.
`repr` should show alternate names and `str` not IMO

**Comment by yunruse** [comment]
Additionally,
- `__repr_code__` should just be `repr_code`,
- `str(meter)` should just return `"meter"`

**Comment by yunruse** [comment]
we should also expect no `*` in `str()`:
```py
>> str(K * m)
'1 K m'
```

**Comment by yunruse** [comment]
Fixed for v1.0.
